engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/escalation/EscalationEventTest.java

```diff
@@ -317,6 +314,155 @@ public void shouldThrowEscalationEventHierarchical() {
+                .limitToProcessInstanceCompleted())
+        .extracting(r -> r.getValue().getBpmnElementType(), Record::getIntent)
+        .containsSubsequence(
+            tuple(BpmnElementType.BOUNDARY_EVENT, ProcessInstanceIntent.ELEMENT_COMPLETING),
+            tuple(BpmnElementType.BOUNDARY_EVENT, ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple(BpmnElementType.CALL_ACTIVITY, ProcessInstanceIntent.ELEMENT_COMPLETING),
+            tuple(BpmnElementType.CALL_ACTIVITY, ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETING),
+            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));
+
+    final var childProcessInstanceKey =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withParentProcessInstanceKey(parentProcessInstanceKey)
+            .getFirst()
+            .getValue()
+            .getProcessInstanceKey();
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_COMPLETED)
+                .withProcessInstanceKey(childProcessInstanceKey)
+                .limitToProcessInstanceCompleted())
+        .extracting(r -> r.getValue().getBpmnElementType())
+        .containsSubsequence(
+            BpmnElementType.INTERMEDIATE_THROW_EVENT,
+            BpmnElementType.MANUAL_TASK,
+            BpmnElementType.END_EVENT);
+
+    assertIsEscalated(childProcessInstanceKey, CATCH_ELEMENT_ID, THROW_ELEMENT_ID, ESCALATION_CODE);
+  }
+
+  @Test
+  public void shouldCatchEscalationOnEscalationStartEventWhenThrowEscalationEventHierarchical() {
+    // given
+    final var processChild =
+        Bpmn.createExecutableProcess("wf-child")
+            .startEvent()
+            .subProcess(
+                "subprocess",
+                s ->
+                    s.embeddedSubProcess()
+                        .eventSubProcess(
+                            "escalation-event-subprocess",
+                            e ->
+                                e.startEvent(CATCH_ELEMENT_ID)
+                                    .interrupting(false)
+                                    .escalation(ESCALATION_CODE)
+                                    .manualTask(TASK_ELEMENT_ID)
+                                    .endEvent())
+                        .startEvent()
+                        .intermediateThrowEvent(
+                            THROW_ELEMENT_ID, b -> b.escalation(ESCALATION_CODE))
+                        .endEvent())
+            .endEvent()
+            .done();
+
+    final var processParent =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .callActivity("call", c -> c.zeebeProcessId("wf-child"))
+            .boundaryEvent("boundary-catch", b -> b.escalation(ESCALATION_CODE))
+            .cancelActivity(false)
+            .endEvent()
+            .done();
```

Comment:
🔧 I think there was a misunderstanding between us about the testcase I was missing. What I meant was this:

We have a parent process with a call activity. The call activity does **not** have a boundary event. Instead the parent process contains an event sub process which will catch the escalation.
The child process that is called by the call activity will simply throw an escalation.

E.g.
Parent:
<img width="343" alt="image" src="https://user-images.githubusercontent.com/5787702/200781176-e5b303b1-09ef-401a-9506-d539a86fa2d0.png">

Child:
<img width="320" alt="image" src="https://user-images.githubusercontent.com/5787702/200781245-01418fca-f1cb-4089-86ba-3f0c1089b71d.png">


This is different from what you did here, as the event sub process is part of the child process in the current  case.